### PR TITLE
C++: Add FP for `cpp/invalid-pointer-deref`

### PR DESCRIPTION
**Pull request by MathiasVP**
I haven't investigated what's going on yet, but this test is a minified version of a result I was seeing on MRVA.

### CODE REVIEW

File: cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp [hunk 1]
```diff
@@ -222,3 +222,34 @@ void test14(unsigned long n, char *p) {
...
+void *realloc(void *, unsigned);
+
+void test17(unsigned *p, unsigned x, unsigned k) {
+    if(k > 0 && p[1] <= p[0]){
+        unsigned n = 3*p[0] + k;
+        p = (unsigned*)realloc(p, n);
+        p[0] = n;
+        unsigned i = p[1];
+        // The following access is okay because:
+        // n = 2*p[0] + k >= p[0] + k >= p[1] + k > p[1] = i
```
Comment (geoffw0) [inline cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp:252]: This is complicated!  I think it's right, except:
```suggestion
        // n = 3*p[0] + k >= p[0] + k >= p[1] + k > p[1] = i
        // (where p[0] denotes the original value for p[0])
```

**Review by geoffw0 — DISMISSED** [review]

File: cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp (hunk 1 above)
Comment (MathiasVP) [reply cpp/ql/test/experimental/query-tests/Security/CWE/CWE-193/pointer-deref/test.cpp:252]: Thanks. You're absolutely right. And yeah, it took me a while to figure out why this was okay as well 🙂

**Comment by MathiasVP** [comment]
> Sorry, my suggestion has probably broken your `.expected` line numbers.

No worries. I've fixed them in e46c53af1d96c65f56750fb751f34886c7e54ead now.

**Review by geoffw0 — APPROVED** [review]

**Comment by MathiasVP** [comment]
Cowboy merging since CI was happy before the test changes 🤠

**Comment by geoffw0** [comment]
🌵 🤠 🌵

**Pull request merged**